File: src/main/java/io/vproxy/vfx/manager/font/FontSettings.java
package io.vproxy.vfx.manager.font;

import javafx.scene.text.Font;
import javafx.scene.text.FontPosture;
import javafx.scene.text.FontWeight;

public class FontSettings {
    public String family;
    public double size = -1;
    public FontWeight weight;
    public FontPosture posture;

    public FontSettings() {
    }

    public FontSettings setFamily(String family) {
        this.family = family;
        return this;
    }

    public FontSettings setSize(double size) {
        this.size = size;
        return this;
    }

    public FontSettings setWeight(FontWeight weight) {
        this.weight = weight;
        return this;
    }

    public FontSettings setPosture(FontPosture posture) {
        this.posture = posture;
        return this;
    }

    public Font build() {
        var family = this.family;
        if (family == null) family = FontManager.FONT_NAME_Default;
        return Font.font(family, weight, posture, size);
    }
}

File: src-ts/src/manager/font/FontSettings.ts
import { FontManager } from './FontManager.js';

export type FontWeight =
  | 'normal'
  | 'bold'
  | 'lighter'
  | 'bolder'
  | 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900;

export type FontPosture = 'normal' | 'italic' | 'oblique';

export class FontSettings {
  family: string | null = null;
  size: number = -1;
  weight: FontWeight | null = null;
  posture: FontPosture | null = null;

  setFamily(family: string): this {
    this.family = family;
    return this;
  }
  setSize(size: number): this {
    this.size = size;
    return this;
  }
  setWeight(weight: FontWeight): this {
    this.weight = weight;
    return this;
  }
  setPosture(posture: FontPosture): this {
    this.posture = posture;
    return this;
  }

  build(): string {
    const family = this.family ?? FontManager.FONT_NAME_Default();
    const style = this.posture ?? 'normal';
    const weight = this.weight ?? 'normal';
    const size = this.size > 0 ? this.size : 16;
    return `${style} ${weight} ${size}px ${family}`;
  }
}
